internal/agent.go

```diff
@@ -58,6 +62,7 @@ func NewAgent(config *Config) (*Agent, error) {
 	ctx, ctxCancel := context.WithCancel(context.Background())
```

Comment:
Could we think of adding the context as parameter to the constructor function? Maybe we can tweak and wire all the context in the entrypoint and pass a single context when we need it during the execution